### ISSUE TITLE
Vague "server could not find the requested resource" message

### ISSUE THREAD

**Issue by jpds** [fluxcd/flux#2120]
I temporarily removed the `metrics-server` deployment from our Kubernetes cluster and saw this appear in our flux logs:

```
flux/flux-7fcb7dd6c4-d9dpd[flux]: ts=2019-06-03T08:05:00.54547656Z caller=loop.go:90 component=sync-loop err="collating resources in cluster for sync: the server could not find the requested resource"
```

It'd be nice if the log message included what it couldn't find. :)

**Additional context**
Add any other context about the problem here, e.g 
- Flux version: 1.12.3
- Kubernetes version: 1.11
- Git provider: Github

**Comment by 2opremio** [comment]
That error is not very helpful. However I am not sure it's directly related to the `metrics-server`.

Did the error only appear once?

**Comment by jpds** [comment]
Yep, only saw it after yanking `metrics-server` and not since.

**Comment by ksaritek** [comment]
if no one is looking at that issue, I am a volunteer for that.

**Comment by hiddeco** [comment]
@ksaritek given the size / impact of this bug, this one is all yours, as we (core maintainers) have some bigger issues to tackle first.

**Comment by ksaritek** [comment]
Hi @jpds, is that metrics-server located in your config repo? 
I tested against minikube version: v1.2.0 and followed the steps at get-started-developing.md

At first:
* k8s: v1.15
* flux: 1.13.3 (master branch)

I enabled metrics-server first (minikube addons enable metrics-server). Then deploy flux and remove metrics-server. After that, I checked the logs but I did not come across the error. 

Then :
* k8s: v1.11.0
* flux: 1.12.3
I enabled metrics-server first (minikube addons enable metrics-server). Then deploy flux and remove metrics-server. After that, I checked the logs but I did not come across the error. 

Also that "the server could not find the requested resource" seems like an error message that returns from k8s API. So I checked to be sure that there are any place that modifying k8s error and seems like no. 

@hiddeco, any idea to do next?
